### PR TITLE
feat(cloudflare-workers-bindings-extension): list all binding types

### PR DESCRIPTION
**Pull request by edmundhung**
Fixes [DEVX-1472](https://jira.cfdata.org/browse/DEVX-1472).

This PR updates the bindings view to list all binding types.

Screenshot 2024-12-19 at 14 39 52

---

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: no e2e tests
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: unreleased

### CODE REVIEW

File: packages/cloudflare-workers-bindings-extension/src/show-bindings.ts [hunk 1]
```diff
@@ -1,30 +1,57 @@
...
-type Node =
-	| {
-			type: "env";
-			config: Config;
-			env: string | null;
-	  }
-	| {
+export type Environment = Required<Config>["env"][string];
+
+export type BindingType =
```
Comment (emily-shen) [inline packages/cloudflare-workers-bindings-extension/src/show-bindings.ts:10]: What about tail consumer bindings?
Should we add anything to https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/docs/how-to/add-a-binding.md

I don't suppose we can share anything from wrangler without it being a massive pain...?

File: packages/cloudflare-workers-bindings-extension/src/show-bindings.ts [hunk 2]
```diff
@@ -198,6 +497,65 @@ export class BindingsProvider implements vscode.TreeDataProvider<Node> {
 	}
 }
 
+export function getBindingName(type: BindingType): string {
```
Comment (emily-shen) [inline packages/cloudflare-workers-bindings-extension/src/show-bindings.ts:500]: nit: could be a record like [`friendlyBindingNames`](https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/utils/print-bindings.ts) in here? 
(very much personal preference)

File: packages/cloudflare-workers-bindings-extension/src/show-bindings.ts (hunk 1 above)
Comment (edmundhung) [reply packages/cloudflare-workers-bindings-extension/src/show-bindings.ts:10]: [f984f60](https://github.com/cloudflare/workers-sdk/pull/7597/commits/f984f60563e8571c37d4ead3a606a14930ce184e)

Added logfwdr, pipelines & send_email. I left `data_blobs`, `text_blobs`, `wasm_modules` out from the implementation as they are legacy (as seen in the comment of the config type) and aren't available on enviornment.

File: packages/cloudflare-workers-bindings-extension/src/show-bindings.ts (hunk 2 above)
Comment (edmundhung) [reply packages/cloudflare-workers-bindings-extension/src/show-bindings.ts]: [54db155](https://github.com/cloudflare/workers-sdk/pull/7597/commits/54db155295c182ea0c22822bbe82b552c5482927)

**Pull request merged**